components/kyma-environment-broker/internal/orchestration/manager/manager.go

```diff
@@ -223,6 +224,43 @@ func (m *orchestrationManager) NewOperationForPendingRetrying(o *internal.Orches
 	return result, o, len(fileterRuntimes), nil
```

Comment:
These should be moved out from `NewOperationForPendingRetrying`, so that they do n't get executed when in `waitForStart`. They should be included in manager.Execute(), after `waitForStart`.


```
if o.Parameters.Kyma == nil || o.Parameters.Kyma.Version == "" {
		o.Parameters.Kyma = &orchestration.KymaParameters{Version: m.kymaVersion}
	}
	if o.Parameters.Kubernetes == nil || o.Parameters.Kubernetes.KubernetesVersion == "" {
		o.Parameters.Kubernetes = &orchestration.KubernetesParameters{KubernetesVersion: m.kubernetesVersion}
	}
	if len(fileterRuntimes) != 0 {
		o.State = orchestration.InProgress
	} else {
		o.State = orchestration.Succeeded
	}
```

Reply:
This part of codes is also used in waitForCompletion() polling loop, judging if there is still runtime left in each loop by len(fileterRuntimes). Do you think we should refactor waitForCompletion() as well?

Reply:
This part of codes is also used in waitForCompletion() polling loop, judging if there is still runtime left in each loop by len(fileterRuntimes). Do you think we should refactor waitForCompletion() as well?

Reply:
This part is not relevant in waitforCompletion(), feel free to extract it and move it to `manager.Execute()`, after `waitForStart`

Reply:
moved first part of them to `manager.Execute()` and kept

```
if len(fileterRuntimes) != 0 {
	o.State = orchestration.InProgress
} else {
	o.State = orchestration.Succeeded
}
```
did many tests and confirmed that when retrying failed orchestration, orchestration state can't be switched correctly without this part.

Reply:
But if we assign `o.State` something different than `Pending` here, then subsequent calls in waitForStart -> resolveOperations loop would not hit the correct branch for Pending:

`if o.State == orchestration.Pending {`

Why failed orchestration retrying didn't work if you moved this part right after the others in `Execute()`, like this:

```
	if o.Parameters.Kyma == nil || o.Parameters.Kyma.Version == "" {
		o.Parameters.Kyma = &orchestration.KymaParameters{Version: m.kymaVersion}
	}
	if o.Parameters.Kubernetes == nil || o.Parameters.Kubernetes.KubernetesVersion == "" {
		o.Parameters.Kubernetes = &orchestration.KubernetesParameters{KubernetesVersion: m.kubernetesVersion}
	}
	if len(operations) != 0 {
		o.State = orchestration.InProgress
	} else {
		o.State = orchestration.Succeeded
	}
```